documentation/concepts/commitments.rst

```diff
@@ -0,0 +1,179 @@
+        _type="each"
+    )
+
+The scheduler sets up such commitments (site-level and device-level) and, together with any prior commitments, hands them to the linear optimizer.
+
+Examples (commitments commonly derived from flex-context)
+--------------------------------------------------------
+
+The examples below map the most common `flex-context` semantics to the
+commitments the scheduler constructs.
+
+1. **Energy (tariff)**
+
+   - *Fields used*: ``consumption-price``, ``production-price``.
+   - *Commitment*: Flow commitment named ``"energy"`` with zero baseline and
+     the two price series as upwards/downwards deviation prices.
+
+2. **Peak consumption**
+
+   - *Fields used*: ``site-peak-consumption`` (baseline) and ``site-peak-consumption-price`` (upwards-deviation price); the downwards price is set to ``0``.
+   - *Commitment*: Flow commitment named ``"consumption peak"``; positive baseline
+     values denote the prior consumption peak associated with sunk costs, and the upwards price penalises going beyond that baseline.
+
+3. **Peak production / peak feed-in**
+
+   - *Fields used*: ``site-peak-production`` (baseline) and ``site-peak-production-price`` (downwards-deviation price); the upwards price is set to ``0``.
+   - *Commitment*: Flow commitment named ``"production peak"``; negative baseline
+     values denote the prior production peak associated with sunk costs, and the downwards price penalises going beyond that baseline.
+
+4. **Consumption capacity**
+
+   - *Fields used*: ``site-consumption-capacity`` (baseline), and ``site-consumption-breach-price`` (upwards-deviation price); the downwards price is set to ``0``.
+   - *Commitment*: Flow commitment named ``"consumption breach"``; positive baseline
+     values denote the allowed consumption limit and the upwards price penalises going
+     beyond that limit.
+
+5. **Production capacity**
+
+   - *Fields used*: ``site-production-capacity`` (baseline) and ``site-production-breach-price`` (downwards-deviation price); the upwards price is set to ``0``.
+   - *Commitment*: Flow commitment named ``"production breach"``; negative baseline
+     values denote the allowed production limit and the downwards price penalises going
+     beyond that limit.
+
+6. **SOC minima / maxima (storage preferences)**
+
+   - *Fields used*: ``soc-minima``, ``soc-minima-breach-price``, ``soc-maxima`` and ``soc-maxima-breach-price``.
+   - *Commitment*: StockCommitment(s) that price deviations below minima or
+     above maxima. Hard storage capacities are set through ``soc-min`` and ``soc-max`` instead and are modelled as Pyomo constraints.
+
+7. **Power bands per device**
+
+   - *Fields used*: ``consumption-capacity`` and ``production-capacity`` (baselines), ``consumption-breach-price`` (upwards-deviation price, with 0 downwards) and ``production-breach-price`` (downwards-deviation price, with 0 upwards).
+   - *Commitment*: FlowCommitment with either baseline and corresponding prices.
+
```

Comment:
I suggest that we add a new section here: "How you can use commitments: An example"

Then provide a flex context example with some price blocks maybe - whatever comes off easy.

(so far, the examples are only from the internal use of commitments)